contracts/orbital-core/src/state.rs

```diff
@@ -1,21 +1,26 @@
-use std::collections::HashMap;
-
 use cosmwasm_schema::cw_serde;
 use cosmwasm_std::{Addr, Uint64};
-use cw_storage_plus::Map;
+use cw_storage_plus::{Item, Map};
+
+pub const USER_NONCE: Item<Uint64> = Item::new("user_nonce");
 
 /// map of users with their respective configurations
-pub const USER_CONFIGS: Map<Addr, UserConfig> = Map::new("user_configs");
+pub const USER_CONFIGS: Map<String, UserConfig> = Map::new("user_configs");
```

Comment:
It is never used for any address-related operations, only for accessing the config. addr check happens before saving it so I felt like it's easier to just store it as a str without addr overhead. e.g. if we query a user config we would need to do an `Addr::unchecked` conversion first.
not really a strong preference on my end though, let me know if you think addr would be better suited here.

Reply:
I think addr should be used inside the contract for addresses on that same chain, the only place where we should not use Addr, is when we get an address from the user (UI / API), and thats only because CW will use `Addr::unchecked` instead of `validate_addr`, and when we are talking about addresses on other chains, because again, we can't use `validate_addr` here.
If the address is the user on that same chain, we should always use `validate_addr` and use `Addr`, not sure why, but you would use `validate_addr` in the query as well.